### DISCUSSION TITLE
Static rendering of a static route question

### DISCUSSION THREAD

**Discussion by lsli8888** [vercel/next.js#69551] [App Router] [answered]
I'm new to next.js and am struggling to statically render a single static route.  I'm probably overlooking a very simple solution.

In my case, I'm trying to statically render a page that lists the top ten most recent blogs.  I want to statically render this page at build time as there's no real need to dynamically serve this blog list as it does not change frequently.  The blogs are saved on my server file system under the src/content/blog directory and are mdx files.  I'm using the mdx and front-matter libraries (not that it really matters)  The blogs are loaded at build time using the node file system module.

Here's my current code (src/app/blog/page.tsx) that dynamically serves the blog list:

```
async function getPosts(): Promise<IPostItem[]> {
  const postsDirectory = path.join(process.cwd(), 'src/content/blog');
  const filenames = fs.readdirSync(postsDirectory);

  const posts = filenames.map((filename: string) => {
    const filePath = path.join(postsDirectory, filename);
    const fileContents = fs.readFileSync(filePath, 'utf8');
    const { data, content } = matter(fileContents);

    return {
      id: '',
      title: '',
      description: '',
      author: {
        id: '',
        name: '',
        avatarUrl: '',
      },
      createdAt: '',
      tags: [],
    };
  });

  return posts;
}

export default async function Page() {
  const posts = await getPosts();
  return <PostListView posts={posts} />;
}
```

Executing `npm run build` indicates this particular route (src/app/blog) is: `(Dynamic)  server-rendered on demand`.

All the examples I see out there require using generateStaticParams with dynamic routes which doesn't fit my scenario.  Other examples also use the fetch API, which doesn't fit in my case as my blog posts are retrieved using the node file system module.  So how can i make this static route served statically?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Are you using, perhaps, `export const dynamic = 'force-dynamic'` or something like that? If not, try with `export const dynamic = 'force-static'`

Though, something else must be at play here, because there are ways to opt-in an entire segment into dynamic behaviour. What does, `PostListView` do?

**Comment by lsli8888** [comment]
Thanks @icyJoseph.  I was not using `export const dynamic = 'force-dynamic`.  I tried `export const dynamic = 'force-static'` and that forced the page to be prerendered with static content.  This is a great improvement!

However, it's not prerendered as static HTML which I was hoping for as I also tried `export const revalidate = 3600` in my code to force regeneration of the static content periodically.  I'll look into this some more.  Thanks!
